src/module/token.ts

```diff
@@ -34,36 +34,38 @@ export class LancerTokenDocument extends TokenDocument {
 export class LancerToken extends Token {
   constructor(document: LancerTokenDocument) {
     super(document);
-    this._spaces = new Set<Point>();
+    this._spaces = [];
   }
 
   /**
    * Cached occupied spaces
    */
-  protected _spaces: Set<Point>;
+  protected _spaces: Point[];
 
   /**
    * Returns a Set of Points corresponding to the grid space center points that
    * the token occupies.
    */
-  getOccupiedSpaces(): Set<Point> {
-    if (this._spaces.size === 0 && canvas?.grid?.type !== CONST.GRID_TYPES.GRIDLESS) {
+  getOccupiedSpaces(mode: "current position" | "updated position" = "current position"): Point[] {
+    let pos: { x: number, y: number } = mode == "current position" ? this.position : this.data;
+
+    if (this._spaces.length === 0 && canvas?.grid?.type !== CONST.GRID_TYPES.GRIDLESS) {
       let hitBox: PIXI.IHitArea;
       if (this.hitArea instanceof PIXI.Polygon) {
         let poly_points: number[] = [];
         for (let i = 0; i < this.hitArea.points.length - 1; i += 2) {
-          poly_points.push(this.hitArea.points[i] + this.position.x, this.hitArea.points[i + 1] + this.position.y);
+          poly_points.push(this.hitArea.points[i] + pos.x, this.hitArea.points[i + 1] + pos.y);
         }
         hitBox = new PIXI.Polygon(poly_points);
       } else if (this.hitArea instanceof PIXI.Rectangle) {
-        hitBox = new PIXI.Rectangle(this.position.x, this.position.y, this.hitArea.width, this.hitArea.height);
+        hitBox = new PIXI.Rectangle(pos.x, pos.y, this.hitArea.width, this.hitArea.height);
       } else {
         // TODO, handle all possible hitArea shapes
         return this._spaces;
```

Comment:
This should probably be `return Array.from(this._spaces)` to avoid consumers accidentally mutating the cache.  Same with the final return.

Reply:
Good point!